src/event.rs

```diff
@@ -59,12 +59,76 @@ impl FfmpegOutput {
   }
 }
 
+/// Represents metadata about a stream.
 #[derive(Debug, Clone, PartialEq)]
-pub struct AVStream {
-  /// Typically `video` or `audio`, but might be something else like `data` or `subtitle`.
-  pub stream_type: String,
-  /// Corresponds to stream `-f` parameter, e.g. `rawvideo`, `h264`, or `mpegts`
+pub struct Stream {
+  /// Corresponds to stream `-f` parameter, e.g. `rawvideo`, `h264`, `opus` or `srt`.
   pub format: String,
+  // The language of the stream as a three letter code such as `eng`, `ger` or `jpn`.
+  pub language: String,
+  /// The index of the input or output that this stream belongs to.
+  pub parent_index: usize,
+  /// The index of the stream inside the input.
+  pub stream_index: usize,
+  /// The stderr line that this stream was parsed from.
+  pub raw_log_message: String,
+  // Data that is specific to a certain stream type.
+  pub type_specific_data: TypeSpecificData,
+}
+
+impl Stream {
+  pub fn is_audio(&self) -> bool {
+    matches!(self.type_specific_data, TypeSpecificData::Audio(_))
+  }
+  pub fn is_subtitle(&self) -> bool {
+    matches!(self.type_specific_data, TypeSpecificData::Subtitle())
+  }
+  pub fn is_video(&self) -> bool {
+    matches!(self.type_specific_data, TypeSpecificData::Video(_))
+  }
+  pub fn is_other(&self) -> bool {
+    matches!(self.type_specific_data, TypeSpecificData::Other())
+  }
+
+  pub fn audio_data(&self) -> AudioStream {
+    if let TypeSpecificData::Audio(audio_stream) = &self.type_specific_data {
+      audio_stream.clone()
+    } else {
+      panic!("This is not an audio stream! Check `stream.is_audio()` before calling this method.");
+    }
+  }
+  pub fn video_data(&self) -> VideoStream {
+    if let TypeSpecificData::Video(video_stream) = &self.type_specific_data {
+      video_stream.clone()
```

Comment:
This is a nit and works just fine either way, but perhaps we can also return a borrowed `&VideoStream` (or `Option<&VideoStream>`) and let the user decide whether  they want to `clone()`. Again taking inspiration from `serde_json` which returns a borrowed value.

Reply:
Yes, the squash & merge button is embedded directly into the merge button itself, which I think only maintainers can trigger.

![image](https://github.com/user-attachments/assets/efe17f44-6277-4dd3-8509-9c59a1ca89f2)